src/main/java/nl/tudelft/semgroup4/GameEndedState.java

```diff
@@ -22,208 +20,406 @@
+     * 
+     * @param container
+     *            container to draw onto
+     */
+    public void drawColumnNames(GameContainer container) {
+        // draws the column names
+        typeFontPlayer.drawString(container.getWidth() / 5 * 2,
+                container.getHeight() / 4, "SCORE", Color.yellow);
+        typeFontPlayer.drawString(container.getWidth() / 5 * 3,
+                container.getHeight() / 4, "MONEY", Color.yellow);
+        typeFontPlayer.drawString(container.getWidth() / 5 * 4,
+                container.getHeight() / 4, "NAME", Color.yellow);
+
+    }
+
+    /**
+     * Draws the continueButton on the screen.
+     * 
+     * @param container
+     *            container to draw onto.
+     */
+    public void drawContinueButton(GameContainer container) {
+        typeFontPlayer.drawString(container.getWidth() - margin
+                - typeFontPlayer.getWidth(gameContinueText),
                 container.getHeight() - margin - typeFontPlayer.getHeight(),
                 "CONTINUE", Color.yellow);
-
     }
 
-    @Override
-    public void update(GameContainer container, StateBasedGame game, int delta)
-            throws SlickException {
-        
-        //checks all mouse clicks in order to be able to select the text boxes and 
-        //continue button.
-        if (input.isMousePressed(Input.MOUSE_LEFT_BUTTON)) {
-            if (mouseOverTextField1.isMouseOver()) {
-                textFieldPlayer1.setFocus(true);
-            } else if (mouseOverTextField2.isMouseOver()) {
-                textFieldPlayer2.setFocus(true);
-            } else if (mouseOverContinueButton.isMouseOver()) {
-                
-                try {
-                    List<HighscoreEntry> highscores = HighscoresHelper.load();
+    /**
+     * Draws a rectangle around the textfield, because slick2D is incapable.
+     * 
+     * @param container
+     *            container to draw onto
+     * @param graphics
+     *            graphics to use to draw the rectangle
+     */
+    public void drawRectangle(GameContainer container, Graphics graphics) {
+        graphics.drawRect(container.getWidth() / 5 * 4,
+                container.getHeight() / 3, textFieldPlayer1.getWidth() - 1,
+                textFieldPlayer1.getHeight() - 1);
 
-                    highscores.add(new HighscoreEntry(
-                            textFieldPlayer1.getText(),
-                            players[0].getScore()));
+    }
 
-                    if (players.length == 2) {
-                        highscores.add(new HighscoreEntry(
-                                textFieldPlayer2.getText(),
-                                players[1].getScore()));
-                    }
+    /**
+     * Sets the fontype of the title.
+     * 
+     * @param typeFontTitle
+     *            the typeFontTitle to set
+     */
+    public void setTypeFontTitle(TrueTypeFont typeFontTitle) {
+        this.typeFontTitle = typeFontTitle;
```

Comment:
I'm not sure, but I think these setters should be protected, if they are only used in tests.